IMS.BL/Program.cs

```diff
@@ -20,53 +23,60 @@ public static void Main(string[] args)
             }           
             
             // Required objects.
+            var dataAccess = new DataAccess(SqlConnectionProvider.Instance.SqlConnectionObject);
             var mapperConfig = new MapperConfig();
-            var mapper = mapperConfig.InitializeAutomapper();
-            var inventory = new Inventory();
-            var inventoryRepository = new InventoryRepository();
+            var inventory = new Inventory(dataAccess);
```

Comment:
you should rename the inventory and call it InventoryService, then uses InventoryRepository inside of it.